lib/models/genres.dart

```diff
@@ -0,0 +1,34 @@
+class Genre {
+  final int id;
+  final String name;
+  static List<Genre> validGenres = [];
+
+  const Genre(
+    this.id,
+    this.name,
+  );
+
+  factory Genre.fromJson(Map<String, dynamic> json) {
+    return Genre(
+      json['id'],
+      json['name'],
+    );
+  }
+
+  static List<Genre>? fromList(List<dynamic>? list) {
+    var result = <Genre>[];
+    list?.forEach(
+      (item) {
+        result.add(Genre.fromJson(item));
+      },
+    );
+    return result;
+  }
+
+  static List<Genre> getGenresFromIDs(List<dynamic> idList) {
+    List<Genre> result =
+        validGenres.where((e) => idList.contains(e.id)).toList();
+
+    return result;
+  }
```

Comment:
This could be moved to another file, Genre should only need to know how to return itself not a Genres list, you can create a class like GenreModel that has a parameter list<Genre> genres

Reply:
Makes sense, I'm guessing this is the difference between Model and Entity (?